fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java

```diff
@@ -672,7 +709,16 @@ private enum BinaryPhysicalOperator {
         GTE_UL(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.UNKNOWN, Type.TypeCode.LONG, (l, r) -> null),
         GTE_UF(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.UNKNOWN, Type.TypeCode.FLOAT, (l, r) -> null),
         GTE_UD(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.UNKNOWN, Type.TypeCode.DOUBLE, (l, r) -> null),
-        GTE_US(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.UNKNOWN, Type.TypeCode.STRING, (l, r) -> null);
+        GTE_US(Comparisons.Type.GREATER_THAN_OR_EQUALS, Type.TypeCode.UNKNOWN, Type.TypeCode.STRING, (l, r) -> null),
+
+        IN_I(Comparisons.Type.IN, Type.TypeCode.INT, Type.TypeCode.ARRAY, InFn::evaluate),
```

Comment:
Who is promoting the input to the correct common type? I.e. `(1, 1.5, 3)` should be promoted to `(1.0, 1.5, 3.0)`.

Reply:
The provided `TypeCode` is the type code of the list item(s), in other words, all list items must share the same type.

In `RelOpValue` we do "implicit" type promotion by the ability to choose a physical operator. The lack of a physical operator represents improper comparison (e.g. `foo` > 5).

I am not sure we can do the same trick here though, as individual array items must "agree" on a common max-type before we can form the in-list during plan generation, for that to happen, we can probably re-use the promotion API you introduced in `insert` code stream. But I think this can be done in a subsequent PR if I am right.

Reply:
We now actually have a way of type promotion for elements in the Array by means of using `AbstractArrayConstructorValue`. It does implicit promotion, if possible, and hence provides with an array of homogeneous type.